data/data/agent/systemd/units/apply-host-config.service.template

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=Service that applies host-specific configuration
+Wants=network-online.target
+Requires=create-cluster-and-infraenv.service
+PartOf=assisted-service-pod.service
+After=network-online.target create-cluster-and-infraenv.service
+ConditionPathExists=/etc/assisted-service/node0
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+Environment=SERVICE_BASE_URL={{.ServiceBaseURL}}
+ExecStartPre=/bin/rm -f %t/%n.ctr-id
+ExecStartPre=/bin/mkdir -p %t/agent-installer /etc/assisted/hostconfig
+ExecStartPre=/usr/local/bin/wait-for-assisted-service.sh
+ExecStart=podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --log-driver=journald --restart=on-failure:10 --pod-id-file=%t/assisted-service-pod.pod-id --replace --name=apply-host-config -v /etc/assisted/hostconfig:/etc/assisted/hostconfig -v %t/agent-installer:/var/run/agent-installer:z --env SERVICE_BASE_URL --env-file /etc/assisted/client_config quay.io/edge-infrastructure/assisted-service:latest /usr/local/bin/agent-installer-client configure
```

Comment:
Given the approach we're taking to create multiple sub-commands for the `agent-installer-client`, what about orchestrating the workflow within a script executed by a single service, instead of having multiple different services with their own dependencies? Especially in those cases where the commands have a good level of cohesion (as it seems here) it could be an occasion to simplify the dependency web (and probably it could also be an help for the testing)

Reply:
I'm sympathetic to that approach. However, having different services allows us to figure out where in the process we are stuck/failing. So I'd like to delay this at least until we have better status reporting.

Reply:
No conflict with updated #6502, so resolving this.